Consturctor parameters don't include fully qualifed names causing compiler error

```
using Ninject;
using System.Collections.Specialized;

namespace Foo
{
    [GenerateFactory]
    public class ExampleClass
    {
        public ExampleClass(OrderedDictionary dictionary) // System.Collections.Specialized
        {

        }
    }
}
```
Will give you the compiler error
`error CS0246: The type or namespace name 'OrderedDictionary' could not be found (are you missing a using directive or an assembly reference?)`. 

This is because the parameters are not using the fully qualified name.

--- a/src/Ninject.Extension.AutoFactories.Sandbox/MixedNamespaceFactory.cs
+++ b/src/Ninject.Extension.AutoFactories.Sandbox/MixedNamespaceFactory.cs
@@ -1,0 +1,19 @@
+﻿using Ninject;
+using System.Collections.Specialized;
+
+namespace Foo
+{
+    /// <summary>
+    /// Test case for https://github.com/ByronMayne/Ninject.Extensions.AutoFactories/issues/14
+    /// Namespaces for parameters were not being written using their fully qualifed name leading to 
+    /// this class being a compiler error.
+    /// </summary>
+    [GenerateFactory]
+    public class ExampleClass
+    {
+        public ExampleClass(OrderedDictionary dictionary) // System.Collections.Specialized
+        {
+
+        }
+    }
+}

--- a/src/Ninject.Extension.AutoFactories.Tests/Snapshots/GenerateFactoryAttributeTests.TypeOfFactory#World.CustomFactory.g.verified.cs
+++ b/src/Ninject.Extension.AutoFactories.Tests/Snapshots/GenerateFactoryAttributeTests.TypeOfFactory#World.CustomFactory.g.verified.cs
@@ -19,7 +19,7 @@ namespace World
 namespace World
 {
 
-    internal sealed partial class CustomFactory : global::World.ICustomFactory 
+    public sealed partial class CustomFactory : global::World.ICustomFactory 
     {
         private readonly global::Ninject.Syntax.IResolutionRoot m_resolutionRoot;
 

--- a/src/Ninject.Extension.AutoFactories/AutoFactorySourceGenerator.cs
+++ b/src/Ninject.Extension.AutoFactories/AutoFactorySourceGenerator.cs
@@ -15,11 +15,9 @@ namespace Ninject.AutoFactories
     [SgfGenerator]
     internal class AutoFactorySourceGenerator : IncrementalGenerator
     {
-        private readonly ProductMapper m_modelMapper;
 
         public AutoFactorySourceGenerator() : base("AutoFactory")
         {
-            m_modelMapper = new ProductMapper();
         }
 
         public override void OnInitialize(SgfInitializationContext context)
@@ -69,7 +67,11 @@ namespace Ninject.AutoFactories
                 .OfType<ConstructorDeclarationSyntax>()
                 .ToArray();
 
-            ProductModel model = m_modelMapper.Map((context.SemanticModel, classDeclaration));
+            ProductMapper mapper = new ProductMapper(context.SemanticModel);
+
+
+
+            ProductModel model = mapper.Map(classDeclaration);
 
             return model;
         }

--- a/src/Ninject.Extension.AutoFactories/Mapping/ConstructorMapper.cs
+++ b/src/Ninject.Extension.AutoFactories/Mapping/ConstructorMapper.cs
@@ -1,4 +1,5 @@
 ﻿using Boxed.Mapping;
+using Microsoft.CodeAnalysis;
 using Microsoft.CodeAnalysis.CSharp.Syntax;
 using Ninject.AutoFactories.Models;
 
@@ -8,9 +9,9 @@ namespace Ninject.AutoFactories.Mapping
     {
         private readonly IMapper<ParameterSyntax, ParameterModel> m_parameterMapper;
 
-        public ConstructorMapper()
+        public ConstructorMapper(SemanticModel semanticModel)
         {
-            m_parameterMapper = new ParameterMapper();
+            m_parameterMapper = new ParameterMapper(semanticModel);
         }
 
         public void Map(ConstructorDeclarationSyntax source, ConstructorModel destination)

--- a/src/Ninject.Extension.AutoFactories/Mapping/ParameterMapper.cs
+++ b/src/Ninject.Extension.AutoFactories/Mapping/ParameterMapper.cs
@@ -1,4 +1,5 @@
 ﻿using Boxed.Mapping;
+using Microsoft.CodeAnalysis;
 using Microsoft.CodeAnalysis.CSharp.Syntax;
 using Ninject.AutoFactories.Models;
 
@@ -6,10 +7,21 @@ namespace Ninject.AutoFactories.Mapping
 {
     internal class ParameterMapper : IMapper<ParameterSyntax, ParameterModel>
     {
+        private readonly SemanticModel m_semanticModel;
+
+        public ParameterMapper(SemanticModel semanticModel)
+        {
+            m_semanticModel = semanticModel;
+        }
+
         public void Map(ParameterSyntax source, ParameterModel destination)
         {
+            SymbolInfo symbolInfo = m_semanticModel
+                .GetSymbolInfo(source.Type!);
+          
+
             destination.Name = source.Identifier.Text;
-            destination.Type = source.Type!.ToString();
+            destination.Type = symbolInfo.Symbol.ToDisplayString();
         }
     }
 }

--- a/src/Ninject.Extension.AutoFactories/Mapping/ProductMapper.cs
+++ b/src/Ninject.Extension.AutoFactories/Mapping/ProductMapper.cs
@@ -2,30 +2,25 @@
 using Microsoft.CodeAnalysis;
 using Microsoft.CodeAnalysis.CSharp.Syntax;
 using Ninject.AutoFactories.Models;
-using System.Diagnostics;
 using System.Diagnostics.CodeAnalysis;
-using System.Linq;
 
 namespace Ninject.AutoFactories.Mapping
 {
-    internal class ProductMapper : IMapper<(SemanticModel, ClassDeclarationSyntax), ProductModel>
+    internal class ProductMapper : IMapper<ClassDeclarationSyntax, ProductModel>
     {
+        private readonly SemanticModel m_semanticModel;
         private readonly ClassAttributeSettings m_classAttributeSettings;
         private readonly IMapper<ConstructorDeclarationSyntax, ConstructorModel> m_constructureMapper;
 
-        public ProductMapper()
+        public ProductMapper(SemanticModel semanticModel)
         {
+            m_semanticModel = semanticModel;
             m_classAttributeSettings = new ClassAttributeSettings();
-            m_constructureMapper = new ConstructorMapper();
+            m_constructureMapper = new ConstructorMapper(semanticModel);
         }
 
-
-
-        public void Map((SemanticModel, ClassDeclarationSyntax) source, ProductModel destination)
+        public void Map(ClassDeclarationSyntax classDeclaration, ProductModel destination)
         {
-            ClassDeclarationSyntax classDeclaration = source.Item2;
-            SemanticModel semanticModel = source.Item1;
-
             AttributeSyntax classAttribute = classDeclaration
               .AttributeLists
               .SelectMany(a => a.Attributes)
@@ -42,7 +37,7 @@ namespace Ninject.AutoFactories.Mapping
 
             if (TryGetFactoryNameArgument(classAttribute, out AttributeArgumentSyntax? factoryNameArgument))
             {
-                switch (factoryNameArgument.GetValue(semanticModel))
+                switch (factoryNameArgument.GetValue(m_semanticModel))
                 {
                     case string asString:
                         fullyQualifedFactoryName = asString;
@@ -72,65 +67,11 @@ namespace Ninject.AutoFactories.Mapping
             if (TryGetMethodNameArgument(classAttribute, out AttributeArgumentSyntax? methodNameArgument))
             {
                 // This will only ever be a string 
-                if (methodNameArgument.TryGetStringValue(semanticModel, out string? parsedValue))
+                if (methodNameArgument.TryGetStringValue(m_semanticModel, out string? parsedValue))
                 {
                     methodName = parsedValue;
                 }
             }
-
-
-
-
-            //if (classAttribute.ArgumentList != null)
-            //{
-            //    // THe constructor can be one of the following
-            //    // 1. ()
-            //    // 2. (typeof(MyType), "MethodName")
-            //    // 3. ("Namespace.MyType", "MethodName");
-            //    AttributeArgumentSyntax[] positionalArgs = classAttribute.ArgumentList.Arguments.OfType<AttributeArgumentSyntax>().ToArray();
-            //    if (positionalArgs.Length > 0)
-            //    {
-            //        object? value = positionalArgs[0].GetValue(semanticModel);
-            //        switch (value)
-            //        {
-            //            case INamedTypeSymbol namedTypeSymbol:
-            //                fullyQualifedFactoryName = namedTypeSymbol.ToDisplayString(NullableFlowState.NotNull);
-            //                switch (namedTypeSymbol.DeclaredAccessibility)
-            //                {
-            //                    case Accessibility.Public:
-            //                        destination.FactoryAccessModifier = AccessModifier.Public;
-            //                        break;
-            //                    case Accessibility.Internal:
-            //                        destination.FactoryAccessModifier = AccessModifier.Internal;
-            //                        break;
-            //                    case Accessibility.Private:
-            //                        destination.FactoryAccessModifier = AccessModifier.Private;
-            //                        break;
-            //                    case Accessibility.ProtectedAndInternal:
-            //                        destination.FactoryAccessModifier = AccessModifier.ProtectedAndInternal;
-            //                        break;
-            //                }
-            //                break;
-            //            case string asString:
-            //                fullyQualifedFactoryName = asString;
-            //                break;
-            //        }
-            //    }
-
-            //    if (positionalArgs.Length > 1)
-            //    {
-            //        if (positionalArgs[1].GetValue(semanticModel) is string asString)
-            //        {
-            //            methodName = asString;
-            //        }
-            //    }
-
-            //    // We can also define just the MethodName 
-            //    if (classAttribute.GetNamedArgumentValue<string>("MethodName", semanticModel, null) is string stringValue)
-            //    {
-            //        methodName = stringValue;
-            //    }
-            //}
 
             destination.FactoryType = new MetadataTypeName(fullyQualifedFactoryName);
 
@@ -155,8 +96,6 @@ namespace Ninject.AutoFactories.Mapping
         /// </summary>
         private bool TryGetFactoryNameArgument(AttributeSyntax attributeSyntax, [NotNullWhen(true)] out AttributeArgumentSyntax? result)
         {
-            result = null;
-
             if (attributeSyntax.TryPositionalArgument(0, out result))
             {
                 return true;

--- a/src/Ninject.Extension.AutoFactories/Templates/FactoryTemplate.cs
+++ b/src/Ninject.Extension.AutoFactories/Templates/FactoryTemplate.cs
@@ -132,7 +132,7 @@ namespace Ninject.AutoFactories.Templates
 
         private void WriteParameter(ClassWriter writer, ParameterModel parameter)
         {
-            writer.Write($"{parameter.Type} {parameter.Name}");
+            writer.Write($"global::{parameter.Type} {parameter.Name}");
         }
     }
 

